recipe/get_plugins.py

```diff
@@ -47,42 +49,96 @@ def checksum(htname, fname):
 
 
 def unzip(fname, dname=None):
+    src_dname, ext = os.path.splitext(fname)
     if dname is None:
-        dname = os.path.splitext(fname)[0]
-
-    with zipfile.ZipFile(fname, "r") as fh:
-        fh.extractall(dname)
+        dname = src_dname
+
+    if ext == ".7z":
+        subprocess.check_call(["7za", "x", "-aos", "-o" + dname, "--", fname])
+    else:
+        with zipfile.ZipFile(fname, "r") as fh:
+            common_prefix_parts = os.path.commonprefix(fh.namelist()).split("/")
+            if len(common_prefix_parts) > 1:
+                # If everything is inside a single top-most folder (e.g., like in GitHub archives),
+                # skip recreating that part of the hierarchy.
+                root_folder = common_prefix_parts[0] + "/"
+                members = []
+                for zip_info in fh.infolist():
+                    new_file_name = zip_info.filename[len(root_folder):]
+                    if new_file_name:
+                        zip_info.filename = new_file_name
+                        members.append(zip_info)
+                fh.extractall(dname, members)
+            else:
+                fh.extractall(dname)
```

Comment:
So when I wrote this I thought, "Hey! We can unzip stuff in Python. :) Plus I have played with `ZipFile` a lot lately. This should be easy." Looking back on it, especially with the `7za` shell out, am wondering if we wouldn't be better off using `7za` for all cases. Thoughts?

Reply:
I only added that part so we don't have to write the commit checksum from the GitHub archives multiple times in `get_plugins.py` and `bld.bat` 😉 .
I just tried `7za` with the additional parameter
* `-spe` : eliminate duplication of root folder for extract command
but that didn't do anything... 😕

Reply:
So `7za` won't work on the zip files?

Reply:
`7za` could extract the zip files but I don't see how we could tell it to skip creating the topmost directory. Currently I think that Python snippet is the most convenient solution.